### ISSUE TITLE
Cancel transaction API error

### ISSUE THREAD

**Issue by simplesoftwaregr** [VivaPayments/API#1307]
When i try to cancel/refund a transaction by making a Delete Http Request at https://www.vivapayments.com/api/transactions/(transactionId)?Amount=(amount)&sourceCode=(sourceCode) with their respective parameters,  i get the following error ""non reversible transaction"

VivaMerchantID = "d64b2507-d74b-4eab-95ca-3bd0a34202a6"
![vivaWallet response](https://user-images.githubusercontent.com/59752679/72260106-bf9cdf00-361a-11ea-88fb-4a9a975c6e97.png)

**Comment by matthewrgourd** [comment]
Sorry to hear that. Can you let me know the payment source ID for which you're attempting a refund?

**Comment by simplesoftwaregr** [comment]
(8291)
Hmm, This is a recurring transaction so i guess this may be the same problem as #1302 isue i posted. Recuring transaction (capture) has no source parametre.

**Comment by simplesoftwaregr** [comment]
I think isue #1302 is fixed. I still get the same error. The payment source is 8291

**Comment by matthewrgourd** [comment]
Hi @simplesoftwaregr, I've checked your account and source. Everything looks fine so I'll need to check with our dev team. Thanks!

**Comment by simplesoftwaregr** [comment]
any news?

**Comment by matthewrgourd** [comment]
Hi @simplesoftwaregr your account shows that you currently have no restriction on API access. Can I ask how the original payment was made, i.e. through which channel?

**Comment by simplesoftwaregr** [comment]
Our payment and cancelation flows work like this:

We send a post HTTP request from our server at https://www.vivapayments.com/api/orders to create a pre-authorization payment order and get a transactionId
Afterwards when we are ready to make the transaction, we send a post HTTP request from our server at https://www.vivapayments.com/api/transactions/{transactionId} with the transactionId we got from the previous step.

This problems occurs on our payment management website, from which a customer can select a transactionId to cancel. In order to cancel a transaction we send a Delete HTTP request from our server at  https://www.vivapayments.com/api/transactions/{transactionId}?Amount={amount}&sourceCode={vivaSourceCode}

**Comment by matthewrgourd** [comment]
Hi @simplesoftwaregr sorry for the delayed reply. Can you send me the transaction ID so I can find it in our system. My email is mgourd@vivawallet.com. If you can reference this issue #1307 when you send the message.

**Comment by matthewrgourd** [comment]
Hi @simplesoftwaregr thanks for sending through the transaction ID. I can't see any obvious reason why the refund wouldn't have worked. For the `sourceCode` parameter, I assume you specified the payment source code where the payment went to originally. How about the amount, is this specified in the smallest denomination of the relevant currency? For example, if you want to refund a payment of £100.37, you need to pass the value "10037".

**Comment by simplesoftwaregr** [comment]
I have figured out the source of the issue. We were using the wrong transactionId on our cancelation requests. After making a post request at  https://www.vivapayments.com/api/transactions/{transactionId} to create a payment, this {transactionId} is what we also try to use on the cancelation requests. Turns out we had to use the transactionId from the response of that HTTP request (As described on the vivaWallet API here) , on the cancelation requests.

Here's the example we have created to solve the issue: We have a transactionId of 528254f7-3b67-4709-87a4-f92956d4016d with Amount : 1 and sourceCode : 8291 . We make the http post request at  https://www.vivapayments.com/api/transactions/528254f7-3b67-4709-87a4-f92956d4016d with the amount and sourceCode on the body to create the payment. In the response of that request we receive the transactionId : af0a6aac-7355-4b4e-9de1-1258a6d420ea. By using that transactionId on a Delete http request at https://www.vivapayments.com/api/transactions/af0a6aac-7355-4b4e-9de1-1258a6d420ea?Amount=1&SourceCode=8291 , we succesfully canceled the transaction. Before we tried to cancel with the original transactionId :  528254f7-3b67-4709-87a4-f92956d4016d
Thank you for your time
